pkg/registry/apis/provisioning/register.go

```diff
@@ -96,6 +99,31 @@ func (b *ProvisioningAPIBuilder) UpdateAPIGroupInfo(apiGroupInfo *genericapiserv
 	if err != nil {
 		return fmt.Errorf("failed to create repository storage: %w", err)
 	}
+
+	repositoryStorage.BeginCreate = b.beginCreate
+	repositoryStorage.BeginUpdate = b.beginUpdate
```

Comment:
^^^ these should likely be the After flavor

two reasons:
1. The comment says [Almost nobody should use this hook.](https://github.com/kubernetes/kubernetes/blob/v1.31.2/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L178)
2. When this becomes an operator, that will execute *after* the values are stored

We already have admission handlers to block things before they happen